lib/active_admin/resource/attributes.rb

```diff
@@ -0,0 +1,30 @@
+module ActiveAdmin
+
+  class Resource
+    module Attributes
+
+      def default_attributes
+        resource_class.columns.each_with_object({}) do |c, attrs|
+          #skip primary key, STI column and counter cache columns
+          unless  c.name == resource_class.primary_key || c.name == resource_class.inheritance_column || c.name =~ /_count$/
+            name = c.name.to_sym
+            attrs[name] = (method_for_column(name) || name)
+          end
+          attrs
```

Comment:
I think you don't need to explicitly return the accumulator when you use `each_with_object`.